### PR TITLE
feat: opt-in preference to auto-watch issues near the user's location

### PR DESCRIPTION
**Pull request by dkastl**
Part of #14 — first of two PRs (this one: preference storage + My account UI; next one: the watcher assignment on issue creation).

## Design (as discussed)

Instead of injecting extra mail recipients, users near a new issue become **watchers** of it — core's notification semantics ("Only for things I watch...") then apply unchanged, following the issue is persistent, and unwatching is the standard opt-out.

## This PR

- **`UserPreferencePatch`**: `gtt_watch_nearby` / `gtt_watch_radius` stored in the serialized `UserPreference#others` hash (no migration), following core's explicit getter/setter pattern so `safe_attributes` mass-assignment from the account form works. `gtt_watch_radius_km` returns a validated positive Integer or nil.
- **My account UI**: a small fieldset rendered directly below the **User Map** via the existing `view_my_account` hook (option B — no Deface into core's Auto watch box). Checkbox + distance (km) number field; both disabled with an explanatory hint until the user has stored a location on the map above.
- **Locales**: en/ja/de, parity checked.

## Tests

- Unit: accessor round trip, `safe_attributes` assignment, truthiness/radius validation edge cases
- Functional (`MyController`): fieldset renders disabled without a location and enabled with one; PUT round trip saves the preference; unchecking turns it off while preserving the stored radius

### CODE REVIEW

File: app/views/redmine_gtt/hooks/_watch_nearby_preference.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+</fieldset>
+
+<fieldset class="box tabular">
+  <legend><%= l(:gtt_label_watch_nearby_legend) %></legend>
+  <% location_set = user.geom.present? %>
+  <p>
+    <label for="pref_gtt_watch_nearby"><%= l(:gtt_label_watch_nearby) %></label>
+    <%= hidden_field_tag 'pref[gtt_watch_nearby]', '0', id: nil %>
+    <%= check_box_tag 'pref[gtt_watch_nearby]', '1', user.pref.gtt_watch_nearby?,
+          id: 'pref_gtt_watch_nearby', disabled: !location_set %>
```
[the remark this answers is not in the document]
Comment (dkastl) [reply app/views/redmine_gtt/hooks/_watch_nearby_preference.html.erb:18]: Fixed in 69bb996: the hidden '0' field is only rendered while the controls are enabled, so with no stored location nothing submits and the preference stays untouched. Added a functional test covering exactly this scenario (pref on + no location + saving other settings).

File: lib/redmine_gtt/patches/user_preference_patch.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      def gtt_watch_nearby?
+        gtt_watch_nearby.to_s == '1'
+      end
+
+      # Validated radius in kilometers: a positive Integer, or nil when the
+      # preference is unset or holds a non-numeric/non-positive value.
+      def gtt_watch_radius_km
+        value = Integer(gtt_watch_radius.to_s, exception: false)
+        value if value&.positive?
+      end
```
[the remark this answers is not in the document]
Comment (dkastl) [reply lib/redmine_gtt/patches/user_preference_patch.rb]: Fixed in 69bb996: `gtt_watch_radius_km` now clamps to `NEARBY_WATCH_MAX_RADIUS_KM` (1000), and the form's `max` attribute references the same constant so they can't drift. Unit test added.
